### ISSUE TITLE
BigQuery UNNEST joins commas are odd

### ISSUE THREAD

**Issue by tunetheweb** [sqlfluff/sqlfluff#4531]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting this SQL wiht default config,, leads to an unexpected comma dedent:

```sql
WITH cte AS (
    SELECT
        table1.col1,
        table1.col2
    FROM
        table1,
        UNNEST(1, 2, 3) AS index_num
    INNER JOIN
        mappings
        ON (index_num)
),

SELECT
    col1,
    col2
FROM
    cte
```

### Expected Behaviour

No change

### Observed Behaviour

Instead you get these errors:

```
$ sqlfluff lint --dialect bigquery test.sql
== [test.sql] FAIL                                                                                                                                                             
L:   6 | P:  15 | LT02 | Expected line break and indent of 4 spaces before ','.
                       | [layout.indent]
L:   7 | P:   1 | LT02 | Expected indent of 12 spaces.
                       | [layout.indent]
L:   8 | P:   1 | LT02 | Expected indent of 8 spaces.
                       | [layout.indent]
L:   9 | P:   1 | LT02 | Expected indent of 12 spaces.
                       | [layout.indent]
L:  10 | P:   1 | LT02 | Expected indent of 12 spaces.
                       | [layout.indent]
All Finished 📜 🎉!

```

And it fixes it to this:

```sql
WITH cte AS (
    SELECT
        table1.col1,
        table1.col2
    FROM
        table1
    ,
        UNNEST(1, 2, 3) AS index_num
    INNER JOIN
        mappings
        ON (index_num)
),

SELECT
    col1,
    col2
FROM
    cte
```

Note the comma after `table1` has moved down

### How to reproduce

See lint above with default config

### Dialect

bigquery

### Version

2.0.0

### Configuration

Default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
This looks to be fixed in latest `main`
